lib/studies/reply/compose_page.dart

```diff
@@ -0,0 +1,281 @@
+import 'package:flutter/material.dart';
+import 'package:gallery/studies/reply/model/email_store.dart';
+import 'package:provider/provider.dart';
+
+class ComposePage extends StatelessWidget {
+  const ComposePage();
+
+  @override
+  Widget build(BuildContext context) {
+    var _senderEmail = 'flutterfan@gmail.com';
+    var _subject = '';
+    var _recipient = 'Recipient';
+    var _recipientAvatar = 'reply/avatars/avatar_0.jpg';
+
+    final emailStore = Provider.of<EmailStore>(context);
+
+    if (emailStore.currentlySelectedEmailId >= 0) {
+      final currentEmail =
+          emailStore.emails[emailStore.currentlySelectedEmailId];
+      _subject = currentEmail.subject;
+      _recipient = currentEmail.sender;
+      _recipientAvatar = currentEmail.avatar;
+    }
+
+    return Scaffold(
+      body: SafeArea(
+        bottom: false,
+        child: SizedBox(
```

Comment:
Same comment as on another PR, you should be able to get away without this, or with `SingleChildScrollView.constraints` or `IntrinsicHeight`

Reply:
Is this what you were thinking of?
```dart
            child: LayoutBuilder(
              builder: (context, constraints) {
                return SingleChildScrollView(
                  child: ConstrainedBox(
                    constraints:
                        BoxConstraints(minHeight: constraints.maxHeight),
                    child: IntrinsicHeight(
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          _MailViewHeader(email: email),
                          const SizedBox(height: 32),
                          _MailViewBody(message: email.message),
                          const SizedBox(height: kToolbarHeight),
                        ],
                      ),
                    ),
                  ),
                );
              },
            ),
```

Reply:
Actually, I like your approach better, it's cleaner. I think a `Container` with `double.infinity` height would also be fine.

Reply:
Sgtm